src/routes/safe/components/Apps/ManageApps.js

```diff
@@ -79,15 +75,23 @@ const ManageApps = ({ appList, onAppAdded, onAppToggle }: Props) => {
 
   const cleanAppInfo = () => setAppInfo(APP_INFO)
 
-  const safeAppValidator = async (value) => {
-    const appInfo = await getAppInfoFromUrl(value)
-
-    if (appInfo.error) {
+  const customRequiredValidator = (value) => {
```

Comment:
Validator with side effects, why?

Reply:
I would prefer validator just to validate things and then we can run the side effect somewhere else depending on its outcome

Reply:
I prefer that too, but this damn lib does not work well with onBlur events.

Reply:
Can this help maybe? https://github.com/final-form/react-final-form-listeners

I remember I took the same approach with changing state inside validator when I was implementing ENS support for address input and I wasn't sure of it, I even talked to lib creator and they suggested me to use the lib I shared.

Reply:
@mikheevm 
The components exposed by that lib are not synced with the form validator. They trigger immediately the input changes. If we make use of it, we need to trigger the validator twice, 1 to update the input message and 1 to check if the input is correct and then to call `setAppInfo`.

Do you prefer that instead of running the side-effect inside the validator? 

I prefer to leave it as it's now, I know that's not the best, but it's pretty "understandable".